bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -532,13 +537,17 @@ private void showTargetTag() {
         }
 
         @Override public void actionPerformed(final ActionEvent actionEvent) {
-            TargetEnvironment env = obsComp.getTargetEnvironment();
-            GuideEnvironment ge = env.getGuideEnvironment();
+            final TargetEnvironment env = obsComp.getTargetEnvironment();
+            final GuideEnvironment ge = env.getGuideEnvironment();
+
+            // REL-2715 demands that we allow manual guide groups to be automatically created if only the auto
+            // group is present. This requires some ugly hacking.
+            final Option<IndexedGuideGroup> iggOpt = positionTable.getSelectedGroupOrParentGroup(env)
+                    .filter(igg -> !igg.group().isAutomatic())
+                    .orElse(() -> appendNewGroup(obsComp, positionTable));
 
-            // Make sure that a group is selected and that it is not the auto group.
-            final Option<IndexedGuideGroup> iggOpt = positionTable.getSelectedGroupOrParentGroup(env);
-            if (!iggOpt.exists(igg -> !igg.group().isAutomatic()))
-                return;
+            // Env may have changed at this point if we appended a new group, so we must re-retrieve env.
+            final TargetEnvironment newEnv = obsComp.getTargetEnvironment();
```

Comment:
This isn't a big deal but I was a bit confused by this at first.  What if the `appendNewGroup` didn't update the data object (which is clearly a side-effect)  but just returned a new target environment and the group it created?  It might help to leave the side effect for the end when totally finished with calculating the new target environment.  In other words, keep it pure until the end at least.